Prepare source tree for pointing GitHub Pages to master docs/

starius opened the pull request:
See https://github.com/mxe/mxe/issues/1500

tonytheodore commented:
Looking good, thanks! A couple of additional comments:
 - `assets` should be moved to `docs/`
 - remove `gh-pages` push in `tools/travis-push.sh` (now or after merge)
 - `docs/release-process.txt` can be removed

tonytheodore commented on src/openmp-validation-1-fixes.patch:
```diff
@@ -42,5 +49,56 @@ diff -u OpenMP3.1_Validation/omp_my_sleep.h OpenMP3.1_Validation.new/omp_my_slee
+Subject: [PATCH] add Batchfile with test to run on host
+
+
+diff --git a/all-tests-openmp-validation.bat b/all-tests-openmp-validation.bat
```
The second patch, `openmp-validation-2-test-crlf.patch`, creates a batch file with crlf line endings. There's special handling for these in `.gitattributes` and I'm not sure it can be consolidated.

tonytheodore replied:
For reasons I don't fully understand, the crlf endings are preserved in the body of the patch and the file is created correctly.

starius replied:
Why should it work incorrectly? We already have many DOS line endings patches created with patch-tool-mxe. File `openmp-validation-2-test-crlf.patch` had mixed line endings: DOS for Copyright header and diff content and Unix for diff header.

Could you explain the problem with DOS line endings and how .gitattributes approaches it, please?

tonytheodore replied:
I can't figure out what the initial problem with DOS line endings was - very likely some local misconfiguration as .gitattributes selectively does what `git config --global core.autocrlf input` does.

starius replied:
`.gitattributes` was added in commit https://github.com/mxe/mxe/commit/b2bef6b006c41ca621470fbe3aa02b6f00848df1 but it does not provide the background.

IMHO, we can stay with consolidated patches, remove `.gitattributes` and revisit the problem if it happens again.

tonytheodore replied:
Agreed.

vog commented:
I believe the automatic translation went slightly too far in the `Makefile` output:

<pre>
 	    @echo 'Warning: Deprecated target name $(1) specified'
 	    @echo
 	    @echo 'Please use $(1).[$(subst $(space),|,$(MXE_LIB_TYPES))] instead'
-	    @echo 'See index.html for further information'
+	    @echo 'See LICENSE.md for licensing information.'
</pre>

If somebody tried to run a deprecated target, we would send them to the actual documentation (`docs/index.html`), not the licensing page, wouldn't we?

vog commented on Makefile:
```diff
@@ -438,7 +437,7 @@ define TARGET_RULE
 	    @echo
 	    @echo 'Please use $(1).[$(subst $(space),|,$(MXE_LIB_TYPES))] instead'
-	    @echo 'See index.html for further information'
+	    @echo 'See LICENSE.md for licensing information.'
```
I believe the automatic translation went slightly too far here.

If somebody tried to run a deprecated target, we would send them to the actual documentation (docs/index.html), not the licensing page, wouldn't we?

starius replied:
Actually it was not automatic, but manual change.
I confused this message with normal copyright text.
Fixed.

tonytheodore replied:
I'll remove this reference in #1499 and replace with a list of valid/replacement targets.

starius commented:
Amended:

  *  assets moved to docs/
  * gh-pages push removed in tools/travis-push.sh
  * docs/release-process.txt removed
  * unneeded change of "See index.html for further information" in Makefile reverted

vog commented:
Note that the Makefile does need a change, though, from "See index.html" to "See docs/index.html".

starius commented:
> Note that the Makefile does need a change, though, from "See index.html" to "See docs/index.html".

Done.

tonytheodore commented:
Thanks @starius, this all looks good and ready to go.

@vog, github thinks there are conflicts but there weren't any on local merge.

starius commented:
Rebased.

@tonytheodore, GitHub seems to be stricter than `git merge`: file build-matrix.html was updated in https://github.com/mxe/mxe/pull/1507 and moved to `docs/` in this pull request. `git rebase` managed to fix this automatically:

```
$ git pull --rebase upstream master 
remote: Counting objects: 11, done.
remote: Compressing objects: 100% (11/11), done.
remote: Total 11 (delta 2), reused 0 (delta 0), pack-reused 0
Unpacking objects: 100% (11/11), done.
From github.com:mxe/mxe
 * branch            master     -> FETCH_HEAD
   46ad578..ac82d04  master     -> upstream/master
First, rewinding head to replay your work on top of it...
Applying: openmp-validation: unify and consolidate the patch
Applying: Copyright headers: point to LICENSE.md and shorten
Applying: rename doc/ to docs/
Applying: mv versions.json build-matrix.html assets docs/
Using index info to reconstruct a base tree...
M       build-matrix.html
M       versions.json
Falling back to patching base and 3-way merge...
Applying: add missing things
Applying: move index.html to docs/
Applying: move CNAME to docs/
Applying: remove docs/release-process.txt
Applying: travis-push.sh: do not push to gh-pages
```

vog commented:
@starius Thanks for your work!
